### PR TITLE
Ensure `doc_stats` are changing even if refresh is disabled

### PR DESCRIPTION
**Pull request by s1monw**
Today if refresh is disabled the doc stats are not updated anymore.
In a bulk index scenario this might cause confusion since even if
we refresh internal readers etc. doc stats are never advancing.
This change cuts over to the internal reader that is refreshed outside
of the external readers refresh interval but always equally `fresh` or
`fresher` which will cause less confusion.

### CODE REVIEW

**Review by bleskes — DISMISSED** [review]
Looks great. A unit tests would be lovely.

**Comment by s1monw** [comment]
@bleskes there was a test, I beefed it up

File: core/src/main/java/org/elasticsearch/index/engine/Engine.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ protected static long guardedRamBytesUsed(Accountable a) {
      * Tries to extract a segment reader from the given index reader.
      * If no SegmentReader can be extracted an {@link IllegalStateException} is thrown.
      */
-    protected static SegmentReader segmentReader(LeafReader reader) {
+    public static SegmentReader segmentReader(LeafReader reader) {
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/engine/Engine.java]: nit: If this utility method get more use-cases, I'd move it to a separate class, like maybe the "Lucene" class in that case?

**Review by jpountz — APPROVED** [review]

File: core/src/main/java/org/elasticsearch/index/engine/Engine.java (hunk 1 above)
Comment (s1monw) [reply core/src/main/java/org/elasticsearch/index/engine/Engine.java]: done, good point

**Pull request merged**